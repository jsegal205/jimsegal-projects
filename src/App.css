body {
  align-items: center;
  background-color: #efefef;
  color: #444444;
  display: flex;
  font-family: Consolas, monaco, monospace;
  height: 100%;
  justify-content: center;
  padding: 0 2em;
}

main {
  flex: 1 1 auto;
  margin-bottom: 4em;
}

.app-header {
  text-align: center;
}

.vote-banner {
  background: #0071bc;
  color: #efefef;
  margin: 0 -2em;
  padding: 2em;
  text-align: center;
}

.vote-banner > a {
  color: #efefef;
}
