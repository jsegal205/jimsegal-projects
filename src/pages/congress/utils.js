export const chamberTitles = (chamber) => {
  switch (chamber) {
    case "house":
      return "Representatives";
    case "senate":
      return "Senators";
    default:
      return "Members";
  }
};

export const individualChamberTitle = (chamber) => {
  const pluralTitle = chamberTitles(chamber);
  return pluralTitle.substring(0, pluralTitle.length - 1);
};

export const otherChamber = (chamber) =>
  chamber === "house" ? "senate" : "house";
