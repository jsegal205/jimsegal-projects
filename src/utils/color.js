const getRGB = (hex) => {
  return {
    r: parseInt(hex.slice(0, 2), 16),
    g: parseInt(hex.slice(2, 4), 16),
    b: parseInt(hex.slice(4, 6), 16),
  };
};

export const randomHexColor = () =>
  Math.floor(Math.random() * 16777215).toString(16);

export const complementaryColor = (hex) => {
  const padZero = (str) => ("00" + str).slice(-2);
  const invertColor = (digit) => padZero((255 - digit).toString(16));
  const { r, g, b } = getRGB(hex);

  return invertColor(r) + invertColor(g) + invertColor(b);
};

export const textColor = (hex) => {
  const { r, g, b } = getRGB(hex);
  const brightness = Math.round((r * 299 + g * 587 + b * 114) / 1000);

  return brightness > 125 ? "444444" : "efefef";
};
