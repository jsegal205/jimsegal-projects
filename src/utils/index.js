// export const apiUrlBase = "https://api.jimsegal.com";
export const apiUrlBase = "http://localhost:8001";

export const projectInfo = [
  {
    absolute: true,
    link: "https://jimsegal.com",
    subtitle: "OG site with other projects",
    title: "JimSegal.com",
  },
  {
    link: "/congress",
    subtitle: "Some fun data surrounding the current congressional sesssion",
    title: "Congress",
  },
  {
    link: "/isJimWearingShorts",
    subtitle: "A site to predict if Jim is wearing shorts or not",
    title: "Is Jim Wearing Shorts?",
  },
  {
    link: "mountaingoat",
    subtitle: "Digital game",
    title: "Mountain Goat Game",
  },
  {
    link: "/recipes",
    subtitle: "A bunch of recipes that I frequent and wanted to show off",
    title: "Recipes",
  },
  {
    link: "/spacex",
    subtitle: "A countdown timer until the next launch of a Space X rocket",
    title: "Space X",
  },
];
