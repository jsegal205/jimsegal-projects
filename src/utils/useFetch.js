import { useState, useEffect } from "react";
import axios from "axios";

// custom hook for performing GET request
const useFetch = (url, initialValue) => {
  const [data, setData] = useState(initialValue);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      try {
        setLoading(true);
        const response = await axios.get(url);
        setData(response.data);
      } catch ({ response }) {
        setData({ status: response.status });
      } finally {
        setLoading(false);
      }
    };
    fetchData();
  }, [url]);
  return { loading, data };
};

export default useFetch;
