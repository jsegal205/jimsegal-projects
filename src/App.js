import React from "react";
import "./App.css";

import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";

import Recipes from "./pages/recipes";
import Recipe from "./pages/recipe";
import Home from "./pages/home";
import NotFound from "./pages/not-found";

export default function App() {
  return (
    <main>
      <h1>Jim Segal Projects</h1>
      <Router>
        <div>
          <nav>
            <ul>
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/recipes">Recipes</Link>
              </li>
            </ul>
          </nav>

          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/recipes" component={Recipes} />
            <Route path="/recipe/:recipeSlug" component={Recipe} />

            <Route path="*" component={NotFound} />
          </Switch>
        </div>
      </Router>
    </main>
  );
}
