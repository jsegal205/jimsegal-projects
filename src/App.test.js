import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "./App";

import { complementaryColor, randomHexColor, textColor } from "./utils/color";

afterEach(cleanup);

jest.mock("./utils/color", () => {
  return {
    complementaryColor: jest.fn(),
    randomHexColor: jest.fn(),
    textColor: jest.fn(),
  };
});

it("renders correctly", () => {
  complementaryColor.mockReturnValue("000000");
  randomHexColor.mockReturnValue("ffffff");
  textColor.mockReturnValue("444444");

  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
