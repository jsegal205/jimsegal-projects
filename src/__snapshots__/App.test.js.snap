// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="app-header"
    >
      Jim Segal Projects
    </h1>
  </header>
  <article>
    <section>
      <ul
        class="project-list"
      >
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="https://jimsegal.com"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                JimSegal.com
              </h2>
            </div>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="/congress"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Congress
              </h2>
            </div>
            <p>
              Some fun data surrounding the current congressional sesssion
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="https://jimsegal.com/IsChicagoColderThanAnchorage/"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Is Chicago Colder Than Anchorage?
              </h2>
            </div>
            <p>
              A way to quickly tell if Chicago, IL is colder than Anchorage, AK
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="/isJimWearingShorts"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Is Jim Wearing Shorts?
              </h2>
            </div>
            <p>
              A site to predict if Jim is wearing shorts or not
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="/mountaingoat"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Mountain Goat Game
              </h2>
            </div>
            <p>
              Digital game
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="https://jimsegal.com/readinglist/"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Reading List
              </h2>
            </div>
            <p>
              Books I have read or listened to
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="/recipes"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Recipes
              </h2>
            </div>
            <p>
              A bunch of recipes that I frequent and wanted to show off
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="/spacex"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Space X
              </h2>
            </div>
            <p>
              A countdown timer until the next launch of a Space X rocket
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="https://jimsegal.com/tabletop/"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Tabletop Games
              </h2>
            </div>
            <p>
              My collection of table top games
            </p>
          </a>
        </li>
        <li
          class="project-list-item"
          style="border-color: #ffffff;"
        >
          <a
            class="project-list-item-link"
            href="https://jimsegal.com/travel/"
          >
            <div
              class="project-list-item-placeholder"
              style="background-color: rgb(0, 0, 0);"
            >
              <h2
                class="project-list-item-title"
                style="color: rgb(68, 68, 68);"
              >
                Travels
              </h2>
            </div>
            <p>
              Places that I have traveled since graduating college in 2008
            </p>
          </a>
        </li>
      </ul>
    </section>
  </article>
</DocumentFragment>
`;
